src/javascript/carouselBox.js

```diff
@@ -0,0 +1,124 @@
+export default function CarouselBox({ target, pagingTarget, prefix, list }) {
+  this.target = target;
+  this.pagingTarget = pagingTarget;
+  this.prefix = prefix;
+  this.list = list;
+  this.leftArrow = null;
+  this.rightArrow = null;
+  this.pagingInterval = null;
+}
+
+CarouselBox.prototype = {
+  init,
+  renderCarouselBox,
+  getCarouselBoxBlock,
+  setTargetInitialStyle,
+  renderArrow,
+  createArrowEvent,
+  moveCarouselBox,
+  shuffleCarouselBoxItem
+}
+
+CarouselBox.constructor = CarouselBox;
+
+function init() {
+  this.renderCarouselBox();
+  this.setTargetInitialStyle();
+  this.renderArrow();
+  this.createArrowEvent();
+}
+
+function renderCarouselBox() {
+  const boxList = this.list.reduce((acc, item) => acc += this.getCarouselBoxBlock(item), '');
+  this.target.insertAdjacentHTML('beforeend', boxList);
+}
+
+function getCarouselBoxBlock(item) {
+  return `
+    <a class="box__item carousel__item" href="">
+      <img class="box__item__img" src="${this.prefix}${item.src}">
+      <span class="box__item__title">${item.title}</span>
+      <span class="box__item__content">${item.content}</span>
+      <span class="box__item__tag">${item.tag}</span>
+    </a>
+  `;
+}
+
+function setTargetInitialStyle() {
+  const { target, list } = this;
+  target.style.width = `${list.length * 20}%`;
+  setTimeout(() => {
+    const childWidth = target.firstElementChild.getBoundingClientRect().width;
+    target.dataset.width = childWidth;
+    target.dataset.default_width = childWidth * -parseInt(list.length / 2);
+    target.style.transform = `translateX(${target.dataset.default_width}px)`;
+  }, 300);
```

Comment:
async-await과 delay함수를 활용해볼 수 있겠습니다.
```suggestion
  await delay(300);
  const childWidth = target.firstElementChild.getBoundingClientRect().width;
  target.dataset.width = childWidth;
  target.dataset.default_width = childWidth * -parseInt(list.length / 2);
  target.style.transform = `translateX(${target.dataset.default_width}px)`;
```

Reply:
> async-await과 delay함수를 활용해볼 수 있겠습니다.

async-await... 찾아보도록 하겠습니다!
이전 과제에서 구현해뒀던 delay를 util 함수로 활용하는 방법도 좋겠네요!
감사합니다 👍 👍